types/scalar_types.go

```diff
@@ -196,6 +197,21 @@ func ValueForType(id TypeID) Val {
 // ParseTime parses the time from string trying various datetime formats.
 // By default, Go parses time in UTC unless specified in the data itself.
 func ParseTime(val string) (time.Time, error) {
+	t, err := parseTimeNonStrict(val)
+	if err != nil {
+		return t, err
+	}
+
+	_, offset := t.Zone()
+	boundary := 23*60*60 + 59*60
+	if offset > boundary || offset < -1*boundary {
```

Comment:
nit: This condition is checked in another place as well. Do we want to export this in a function and call that?

Reply:
Done